docs/source/migration.md

```diff
@@ -0,0 +1,716 @@
+    
+    from neural_compressor.training import prepare_compression, WeightPruningConfig
+    ##setting configs
+    pruning_configs=[
+    {"op_names": ['layer1.*']，"pattern":'4x1'},
+    {"op_names": ['layer2.*']，"pattern":'1x1', 'target_sparsity':0.5}
+    ]
+    configs = WeightPruningConfig(
+        pruning_configs=pruning_configs,
+        target_sparsity=config.target_sparsity,
+        pattern=config.pattern,
+        pruning_frequency=config.pruning_frequency,
+        start_step=config.start_step,
+        end_step=config.end_step,
+        pruning_type=config.pruning_type,
+    )
+    compression_manager = prepare_compression(model=model, confs=configs)
+    compression_manager.callbacks.on_train_begin()  ## insert hook
+    for epoch in range(num_train_epochs):
+        model.train()
+        for step, batch in enumerate(train_dataloader):
+            compression_manager.callbacks.on_step_begin(step) ## insert hook
+            outputs = model(**batch)
+            loss = outputs.loss
+            loss.backward()
+            compression_manager.callbacks.on_before_optimizer_step()  ## insert hook
+            optimizer.step()
+            compression_manager.callbacks.on_after_optimizer_step() ## insert hook
+            lr_scheduler.step()
+            model.zero_grad()
+    ...
+    compression_manager.callbacks.on_train_end()
+```
+
+## Distillation
+
+Distillation is one of popular approaches of network compression, which transfers knowledge from a large model to a smaller one without loss of validity. As smaller models are less expensive to evaluate, they can be deployed on less powerful hardware (such as a mobile device).
+
+**Distillation with Intel Neural Compressor 1.X**
+
+Intel Neural Compressor distillation API is defined under `neural_compressor.experimental.Distillation`, which takes a user yaml file `conf.yaml` as input.
+
+```yaml
+    distillation:
+    train:                    # optional. No need if user implements `train_func` and pass to `train_func` attribute of pruning instance.
+        start_epoch: 0
+        end_epoch: 10
+        iteration: 100
+        
+        dataloader:
+          batch_size: 256
+        criterion:
+          KnowledgeDistillationLoss:
+              temperature: 1.0
+              loss_types: ['CE', 'KL']
+              loss_weights: [0.5, 0.5]
+        optimizer:
+          SGD:
+              learning_rate: 0.1
+              momentum: 0.9
+              weight_decay: 0.0004
+              nesterov: False
+    evaluation:                              # optional. required if user doesn't provide eval_func in neural_compressor.Quantization.
+    accuracy:                              # optional. required if user doesn't provide eval_func in neural_compressor.Quantization.
+        metric:
+        topk: 1                            # built-in metrics are topk, map, f1, allow user to register new metric.
+        dataloader:
+        batch_size: 256
+        dataset:
```

Comment:
how to implement the 'dataloader' item in 2.x API?
Expect to see how to replace every items in yaml by the codes of 2.x API.

Reply:
There is a lot of configuration information in the previous yaml file do not have any function. Not all parameters have their own mapping pattern. We have deleted the unnecessary items in 2.x API. We directly input the dataloader to the train function. I will add a sentence about it. The other information is no more used.